bodhi/server/models.py

```diff
@@ -1529,6 +1564,36 @@ def validate_builds(self, key, build):
             raise ValueError(u'An update must contain builds of the same type.')
         return build
 
+    @validates('release')
+    def validate_release(self, key, release):
+        """
+        Make sure the release is the same content type as this update.
+
+        Args:
+            key (str): The field's key, which is un-used in this validator.
+            release (Release): The release object which is being associated with this update.
+        Raises:
+            ValueError: If the release being associated is not the same content type as the
+                update.
+        """
+        if release and self.content_type is not None:
+            for u in release.updates:
```

Comment:
Do we really want to go over all updates?
I'd say that if one of them is there and of the same type, we can break.

Reply:
Yeah I had briefly wondered about this too when I was doing this. I think I agree. In the long term I plan to formalize the Release and give it a DB enforced type anyway. Fixed.